server/plugin.go

```diff
@@ -410,26 +410,27 @@ func (p *Plugin) restart() {
 
 func (p *Plugin) generatePluginSecrets() error {
 	needSaveConfig := false
-	if p.configuration.WebhookSecret == "" {
+	cfg := p.getConfiguration()
```

Comment:
Let's `Clone` here, as the `configuration` struct should be considered immutable:
```suggestion
	cfg := p.getConfiguration().Clone()
```